fiftyone/server/samples.py

```
@@ -156,24 +197,58 @@ async def paginate_samples(
+            # rather than making an extra query just to guess.
+            has_previous_page=more if backward else False,
+            has_next_page=False if backward else more,
             start_cursor=edges[0].cursor if edges else None,
-            end_cursor=edges[-1].cursor if len(edges) > 1 else None,
+            end_cursor=edges[-1].cursor if edges else None,
         ),
         edges=edges,
     )
 
 
+def _encode_cursor_value(value: t.Any) -> str:
```

Ruff (0.15.15)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)